### DISCUSSION TITLE
Slow App when adding multiple currencies

### DISCUSSION THREAD

**Discussion by Dilosen** [ghostfolio/ghostfolio#3490] [Q&A] [answered]
I am having an issue where the app slows down completely, not working and crashing.  I can only seem to use one currency, which I assume due to default base currency being USD, it is USD. Thus I can only import my USD transactions onto the platform. 

If I try to import any other currency, such as AUD or EUR, and it's respective transactions, the site seems to be in a state of "loading"  as no data is being populated. I am unable to see Analysis, Overview or any other pages in fact. 

This also happens on the "free cloud" web app as well which is confusing, on my ubuntu/umbrel set ups, it crashes completely.

Is it due to currency conversion or getting the rates?

**Comment by dtslvr** [comment]
Hi @moosesnWoop

Thank you for pointing this out. It should be fixed now via https://github.com/ghostfolio/ghostfolio/pull/3489.

Could you please double check the dates of your activities (Hint: `0202-03-13`)?

**Reply by Dilosen** [reply]
Good Day Thomas,  I have pulled the latest version from github. I am using a standalone installation on a self hosted server (old laptop). All my activity dates are formatted as dd/mm/yyyy - is it meant to be something else? (Hint: 0202-03-13)

The issue specifically occurs as soon as I add ANOTHER currency other than the base currency (default: USD) - so far, I have only put about 55 USD transactions with NO issues. 

As soon as I add my AUD transactions, the platfrom grinds to a halt (no loading of anything). I have tried it again, see the screenshot, as the webapp stops/slows as soon as I add another account say, Australian Stock Exchange. I've added about 15 transactions now, 


![ytuy](https://github.com/ghostfolio/ghostfolio/assets/16508683/c0bc10c2-40ab-4572-96f8-e4bc542f78a9)

**Reply by Dilosen** [reply]
Hi, just an update:

I am ghostfolio in docker container. Every time I navigate to a different page there is a large amount of CPU load in the container >100%.  After a few minutes, this may subside, depending on the page or the data that needs to be displayed and will display the data or chart or transactions. 

There are only ~25 transactions, in AUD and USD. As mentioned prior, if one currency there is not load issues. 

The server laptop is running i3 4 cores, 4GB RAM.

**Reply by dtslvr** [reply]
I have discovered a problematic activity with date `0202-03-13` in the cloud. If you have nothing to do with this, please just ignore it 🙂

> There are only ~25 transactions, in AUD and USD. As mentioned prior, if one currency there is not load issues.

Do you see anything noticeable in the server log?

**Reply by Dilosen** [reply, marked as answer]
Hi Thomas, I have done a fresh install with the latest updates and it works much much faster now with multiple currencies. I am having an issue there, but will open another thread for that. 

Thanks for the latest update.
